### PR TITLE
Added case search filter to exclude related cases

### PR DESCRIPTION
**Pull request by orangejenny**
## Summary
Part of https://dimagi-dev.atlassian.net/browse/USH-1252

## Feature Flag
Case search & claim

## Product Description
Prevents invalid results from showing up in case search. Unlikely to be relevant for non-USH projects.

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

Tests included.

### QA Plan

Not requesting QA for this piece. Tested locally that I can still build apps and use case search.

### Safety story
Has decent test coverage, adds a filter that will be ignored until the other part of this ticket is done.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

**Review by esoergel — APPROVED** [review]

File: corehq/apps/app_manager/suite_xml/sections/remote_requests.py [hunk 1]
```diff
@@ -166,6 +166,7 @@ def _build_remote_request_datums(self):
                     additional_types, instance_name=RESULTS_INSTANCE)
             if self.module.search_config.search_filter:
                 nodeset = f"{nodeset}[{interpolate_xpath(self.module.search_config.search_filter)}]"
+        nodeset += "[not(commcare_is_related_case=true())]"
```
Comment (snopoke) [inline corehq/apps/app_manager/suite_xml/sections/remote_requests.py:169]: could make this a constant

**Review by snopoke — APPROVED** [review]

File: corehq/apps/app_manager/suite_xml/sections/remote_requests.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/app_manager/suite_xml/sections/remote_requests.py]: 74c15d6e1aa

**Comment by orangejenny** [comment]
Rebased on top of https://github.com/dimagi/commcare-hq/pull/30324/ since the code is related and https://github.com/dimagi/commcare-hq/pull/30323/commits/74c15d6e1aa688abae2c7890fa7f4a2960f8573e introduced a conflict

**Review by esoergel — APPROVED** [review]

**Comment by orangejenny** [comment]
Cancelled build, which is pre-rebase.

**Pull request merged**